CoffeeShop.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include "LeakedObjectDetector.h"
+#include "CoffeeMachine.h"
+#include "Computer.h"
+
+struct CoffeeShop
+{
+    CoffeeShop();
+    ~CoffeeShop();
+
+    CoffeeMachine coffeeMachine1;
+    Computer pc;
+
+    struct Order
+    {    
+        Order();
+        ~Order();
+        std::string customerName {"new customer"};
+        int brewStrength {3};
+        int numCups {1};
+        std::string drinkName {"Flat White"};
+        JUCE_LEAK_DETECTOR(Order)
+    };
+
+    struct Customer
+    {
+        Customer();
+        ~Customer();
+        Order getPreferredOrder();
+    };
```

Comment:
Give this class a `name` member variable.
modify the constructor to have 1 parameter: `std::string n`.
initialize the `name` member variable with the constructor parameter `n`.
in `getPreferredOrder()`, set the `order.customerName` to this `name` member variable.
- [ ] revise